### PR TITLE
Update "Support for WebAssembly (Wasm)" page

### LINKED ISSUE #11319: Update "Support for WebAssembly (Wasm)" page
- [ ] Ensure instructions for flutter build web --wasm command are clear
- [ ] Remove dhttpd instructions - flutter run works now.
- [ ] JS Interop
	- Use package:web and dart:js_interop, they are replacing dart:html package:js
	- How to tell if a package is Wasm-compatible
	- Flutter community has been updating packages to be Wasm-compatible
- [ ] Remove "Chrome 119 or later" - Chrome is on version 130 now. Point to https://docs.flutter.dev/reference/supported-platforms for supported platforms.

### PR DESCRIPTION
**Pull request by johnpryan**
Resolves #11319 

cc: @kevmoo

### CODE REVIEW

File: src/content/platform-integration/web/wasm.md [hunk 1]
```diff
@@ -79,123 +37,93 @@ that has been migrated to be
...
+$ flutter create . --platforms web
+```
+
 [Flutter web app initialization]: /platform-integration/web/initialization
 
-### Build or run the app `flutter build web --wasm`
+### Run or build your app
+
+To run the app with Wasm for development or testing,
+use the `--wasm` flag with the `flutter run command.
```
Comment (sfshaza2) [inline src/content/platform-integration/web/wasm.md:57]:
```suggestion
use the `--wasm` flag with the `flutter run` command.
```

File: src/content/platform-integration/web/wasm.md [hunk 2]
```diff
@@ -79,123 +37,93 @@ that has been migrated to be
...
-// Using the environment variable set by compilation (preferred)
 const isRunningWithWasm = bool.fromEnvironment('dart.tool.dart2wasm');
-
-// Using differences in number representation to test whether native (Wasm)
-// number representation is used
-final isRunningWithWasm = identical(double.nan, double.nan);
 ```
 
-## Known limitations
+Alternatively, you can use differences in number representations
```
Comment (sfshaza2) [inline src/content/platform-integration/web/wasm.md:85]: As re-written, how is this a limitation? Can you make it more clear?

File: src/content/platform-integration/web/wasm.md [hunk 3]
```diff
@@ -79,123 +37,93 @@ that has been migrated to be
...
+To run a Flutter app that has been compiled to Wasm,
+you need a browser that supports [WasmGC][].
 
-As mentioned in [Load it in a browser](#load-it-in-a-browser),
-to run Flutter web apps compiled to Wasm,
-use _Chrome 119 or later_.
+[Chromium and V8][] released stable support for WasmGC in Chromium 119.
+Note that Chrome on iOS uses WebKit, which doesn't yet [support WasmGC][].
+Firefox announced stable support for WasmGC in Firefox 120,
+but currently doesn't work due to a [known limitation](#known-limitations).
```
Comment (sfshaza2) [inline src/content/platform-integration/web/wasm.md:99]: Not clear what the limitation is.

Comment (johnpryan) [reply src/content/platform-integration/web/wasm.md:99]: Oops, this is a broken link. This is referring to the "Why not Firefox?" section, which I moved directly underneath this paragraph, so I'll remove this link and try to make that more clear.

File: src/content/platform-integration/web/wasm.md (hunk 2 above)
Comment (johnpryan) [reply src/content/platform-integration/web/wasm.md:85]: This section isn't about limitations, it's about detecting if your app is running with WebAssembly. I moved the "known limitations" section further down the page.

**Comment by johnpryan** [comment]
@sfshaza2 can you take another look?

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Comment by krokyze** [comment]
@sfshaza2 I think the part about headers, should had been kept:

```
Cross-Origin-Embedder-Policy: credentialless
Cross-Origin-Opener-Policy: same-origin
```

I spent few hours debugging why WebAssembly (Wasm) wasn’t working in the production build (using the latest version of Chrome) until I added these specific headers.

**Comment by johnpryan** [comment]
https://github.com/flutter/website/pull/11477

**Pull request merged**